src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -480,6 +482,81 @@ public void ascii() throws IOException {
         );
     }
 
+    @Test
+    public void ifTest() throws IOException {
+        assertThat(
+                executeQuery("SELECT IF(age > 30, 'True', 'False') AS Ages FROM " + TEST_INDEX_ACCOUNT
+                        + " WHERE age IS NOT NULL GROUP BY Ages", "jdbc"),
+                containsString("\"type\": \"keyword\"")
+        );
+        assertThat(
+                executeQuery("SELECT IF(2 > 0, 'hello', 'world') AS if FROM " + TEST_INDEX_ACCOUNT),
+                hitAny(kvString("/fields/if/0", equalTo("hello")))
+        );
+        assertThat(
+                getHits(
+                executeQuery("SELECT IF(2 = 0, 'hello') AS if FROM " + TEST_INDEX_ACCOUNT)
+                ).getJSONObject(0).query("/fields/if/0"),
+                equalTo(null)
+        );
+    }
+
+    @Test
+    public void ifnull() throws IOException {
+        assertThat(
+                executeQuery("SELECT IFNULL(lastname, 'unknown') AS name FROM " + TEST_INDEX_ACCOUNT
+                        + " GROUP BY name", "jdbc"),
+                containsString("\"type\": \"keyword\"")
+        );
+        assertThat(
+                executeQuery("SELECT IFNULL('sample', 'IsNull') AS ifnull FROM " + TEST_INDEX_ACCOUNT),
+                hitAny(kvString("/fields/name/0", equalTo("sample")))
+        );
+
+        //TODO: let null pass the verification of semantic analyzer
+//        assertThat(
+//                executeQuery("SELECT IFNULL(null, 'IsNull') AS ifnull FROM " + TEST_INDEX_ACCOUNT),
+//                hitAny(kvString("/fields/name/0", equalTo("IsNull")))
+//        );
+    }
+
+    @Test
+    public void isnull() throws IOException {
+        assertThat(
+                executeQuery("SELECT ISNULL(lastname) AS name FROM " + TEST_INDEX_ACCOUNT + "ORDER BY name", "jdbc"),
```

Comment:
Just wonder what will happen if the field doesn't exist in some document? It is possible in ES so we may want to check if our impl can take care of this.

Reply:
ES returns field[...] cannot be found or used here. DId you mean to precheck the field that doesn't exit before running into ES?

Reply:
Sorry, I meant the field does exist in index mapping but some document doesn't have it. For example, some document doesn't include `lastname` in your test case.